exercises/2/fold.expresion.cpp

```diff
@@ -11,8 +11,8 @@
 #include <iostream>
 template<typename ... T>
 auto average(T ... t) {
-    return (t + ... ) / sizeof...(t);
+    return static_cast<double>((t + ... )) / sizeof...(t);
```

Comment:
我个人建议是模仿 `<cmath>` 中浮点函数的额外重载，把整数类型转换为 `double` 而浮点类型不变。这样可以避免 `long double` 被转换为 `double`。

```suggestion
    constexpr auto integer_to_double = [](auto x) {
         if constexpr (std::is_integral_v<decltype(x)>)
             return static_cast<double>(x);
         else
             return x;
    };
    return (integer_to_double(t) + ... ) / sizeof...(t);
```

Reply:
static_caste() it will explicitly change the data type of t to double
``` c++
#include <iostream>

template<typename ... T>
auto average(T ... t) {
    return (static_cast<double>(t) + ...) / sizeof...(t);
}

int main() {
    std::cout << average(1, 2, 3, 4, 5, -6, -7, -8, -9, -10) << std::endl;
}
}
```